src/mid.cpp

```diff
@@ -941,21 +955,50 @@ void CmidPlayer::rewind(int subsong)
                 adlib_style=LUCAS_STYLE|MIDI_STYLE;
                 //note: no break, we go right into midi headers...
             case FILE_MIDI:
+            case FILE_MIDItype1:
                 if (type != FILE_LUCAS)
                     tins=128;
-                getnext(11);  /*skip header*/
+                /* "Mthd"              header-chunk
+                 * 0x00 0x00 0x00 0x06 header size
+                 * 0x00 0x0n           midi file type: 0=single-track format, 1=multiple-track format, 2=multiple-song format
+                 * 0xnn 0xnn           track-number
+                 * 0xnn 0xnn           tempo
+                 */
+                getnext(7);
+                i=getnext(2);
+                type=(i==1)?FILE_MIDItype1:FILE_MIDI;
+		midiprintf ("type:%ld\n", i);
+                getnext(2); /* skip track-number */
                 deltas=getnext(2);
                 midiprintf ("deltas:%ld\n",deltas);
-                getnext(4);
+
+	        for (i=0; i<16; i++)
+	        {
+                  ch[i].nshift=-13;
```

Comment:
Is this where it does the octave shift?  Could you add a comment to say why this is done?  It just seems like it might be the wrong place to fix the problem, so someone might want to remove it later if they work out where the proper fix should be.

Reply:
(The shift also included one semi-tone). I will try to find some midi files that makes it easier to verify if this is correct or not.

Reply:
(courtesy of https://github.com/cirlabs/miditime for generating test midifile)
[miditest.zip](https://github.com/adplug/adplug/files/8103942/miditest.zip)

The very first note should be a C-4 at 261Hz, while adplug plays it back at 129Hz which is slightly below a C-3.

The -13 is because the default value in the midi playback engine is -25, so increasing one octave == +12 semitones, that gives us -13 instead.

Reply:
That leaves Lucas MIDI files - they must be tested if they should have -25 or -13. They will currently have -13. I do not have the files, nor I do I know how they should sound. Can you test if they are too high now, if so I will add an if() statement

Reply:
This change does increase all notes by one octave in the existing music test file `mi2.laa`, so the test fails. I don't know if this makes it sound more correct, needs to be tested with original player or game.

I think it's better to propose this fix in a separate PR, since it's not really related to detecting MIDI type 1 files.

Reply:
MIDI files are without this at all one semi-tone off. How much is LAA files off, without this patch applied. Perfect or one semi-tone off?

Reply:
I don't have any LAA files to check, judging only by one single file in the tests.

Reply:
Please file a separate issue regarding MIDI files having one octave + one semi-tone off.